extensions/gmsa-dc/v1/Promote-DC.ps1

```diff
@@ -0,0 +1,58 @@
+<#
+ .SYNOPSIS
+ This script will automate the creation of a Forest, Domain, and Domain Controller
+
+ .NOTES
+ This is only for automated e2e testing.  DO NOT use this for production.
+ Jeremy Wood (JeremyWx)
+ Version: 1.0.0.0
+#>
+
+Set-StrictMode -Version Latest
+$ErrorActionPreference = 'Stop'
+$PSDefaultParameterValues['*:ErrorAction'] = 'Stop'
+
+# Create a working directory and change to it
+mkdir C:\gmsa
+Set-Location -Path C:\gmsa
+
+# Logging for troubleshooting
+Start-Transcript -Path "C:\gmsa\Promote.txt"
+
+# Download gMSA Setup Script
+Invoke-WebRequest -UseBasicParsing https://raw.githubusercontent.com/kubernetes-sigs/windows-testing/master/extensions/gmsa-dc/v1/Setup-gMSA.ps1 -OutFile Setup-gMSA.ps1
```

Comment:
Not a fan of this download link to the setup script. Say we ever move this script somewhere else, or someone wants to test it from a fork, remembering that this reference to this exact repo exists here is wishful :). AFAIK, the Azure Custom Script Extension schema defines "fileUris" as an array, so it should mean that we can let the extension download all necessary files. It's an assumption, I've never actually tested it :) That change would be ideal, but if not, at least we should define the URL at the beginning of the script for visibility.